Summation of the minisummit blogpost (#285)
3mdeb/news-and-ideas: blog/content/post/2020-06-15-qubes_summation.md

---
title: Qubes OS & 3mdeb 'minisummit' 2020 summation

> abstract: 'The second Qubes OS & 3mdeb minisummit is ahead of us. We had gone through four evenings of topics devoted to Qubes OS, so it is time for broad summation of the event.'

Author: done

cover: /covers/qubes&3mdeb_logo.png
author: kamila.banecka
layout: post
published: true
date: 2020-06-15
archives: "2020"

tags:
  - QubesOS
  - minisummit
  - live
  - YouTube
  - firmware
  - coreboot
  - security
  - TPM
categories:
  - Miscellaneous
  - Security
  - Firmware

---
The second Qubes OS and 3mdeb 'minisummit' is ahead of us. We have gone through
four evenings of topics devoted to Qubes OS. It has been another fruitful and
productive event, during which we have exchanged loads of knowledge and
expertise. Below you will find list of topics, we have discussed with related
videos and broad report of each day.

> * 20.05

Reviewer: @KBanecka please also add slides for my presentations: https://cloud.3mdeb.com/index.php/f/247810
There are 3 PDFs there. For example:

```
  1. Michał Żygowski: Qubes OS on modern AMD platform
    * [recording](https://www.youtube.com/watch?v=Rw7rAPPyPPc&t=31s)
    * [slides](link)
```

@pietrushnic @Asiderr maybe consider publishing slides?

Reviewer 2: https://www.slideshare.net/PiotrKrl/srtm-for-qubes-os-vms
https://www.slideshare.net/PiotrKrl/drtm-for-qubes-os-vms

Author: I am waiting for Norbert and Marek presentations

  1. Michał Żygowski: Qubes OS on modern AMD platform
    * [recording](https://www.youtube.com/watch?v=Rw7rAPPyPPc&t=31s)
  2. Norbert Kamiński: Status fwupd/LVFS support for Qubes OS
    * [recording](https://www.youtube.com/watch?v=o_IdERo3aiE&t=984s)
  3. AMA session
    * [recording](https://www.youtube.com/watch?v=BSGUcW6QDYU&t=1509s)

* 28.05
  1. Piotr Król: SRTM for Qubes OS VMs
    * [recording](https://www.youtube.com/watch?v=Eip5Rts6S2I&t=2s)
  2. Michał Żygowski: Anti Evil Maid for modern AMD UEFI-based platform
    * [recording](https://youtu.be/rM0vRi6qABE?t=3)
  3. AMA session
    * [recording](https://youtu.be/rM0vRi6qABE?t=1904)

* 04.06
  1. Piotr Król: DRTM for Qubes OS VMs
    * [recording](https://youtu.be/pZF-jyJWTE4)
  2. Michał Żygowski: Anti Evil Maid for Intel coreboot-based platform
    * [recording](https://youtu.be/YE2FbFlszI4?t=9)
  3. AMA session
    * [recording](https://youtu.be/YE2FbFlszI4?t=1725)

> * 10.06

Reviewer 3: @KBanecka do we have recordings already?

Author: I've added recordings.

  1. Frédéric Pierret: How to build Qubes? From components to operating system
overview
  2. Marek Marczykowski-Górecki: Operating system testing, when it itself uses
  virtual machines
  3. AMA session

### Qubes OS future: AMD SEV and fwupd/LVFS

 The first day of 'minisummit' gathered us around Qubes OS future: AMD SEV and
fwupd/LVFS, where prelegents were discussing projects based on Qubes OS. The

> opening presetation was held by Michał Żygowski, Firmware Engineer in 3mdeb,

Reviewer 2: @KBanecka please link also to profiles on this website

Author: done

> opening presetation was held by Michał Żygowski, Firmware Engineer in 3mdeb,

Reviewer 3: @KBanecka `presentation`

Author: corrected

Braswell SoC, PC Engines and Protectli maintainer in coreboot who has presented
Qubes on SuperMicro M11SDV-4C-LN4F superserver platform, that has been tested
with Qubes OS on board. Michał has introduced and explained it’s features,

> Secure Encrypted Visualisation (SEV) and Secure Memory Encryption (SME) with

Reviewer: ```suggestion
Secure Memory Encryption (SME) and Secure Encrypted Visualisation (SEV) with
```

Author: done

encrypted state (SEV-ES) extension and explained why SME is better than TSME.
Afterwards, we have been introduced with status of mentioned features on Open

> Source, debating whether it is suppoorted, whether it can be used, what impact

Reviewer 3: @KBanecka `supported`

Author: corrected

Qubes OS will have on the features or is it possible to run SEV-enabled guests
using open-source tools. Michał described briefly also OVMF and SUSE Linux
Enterprise Server support for SEV, Automatic Exits (AE) and Non-automatic Exits
(NAE). After prelection, our specialists took the floor debating over host
system protection, memory encryption on the BIOS level, XEN support and many

> more issues that can be found in the video above.

Reviewer 2: @KBanecka I think we should consider 2 things for each presentation

- how community can engage
- what we hope to do in next 12 months

I believe you can ask authors of presentations

The second prelegent was Norbert Kamiński, Junior Embedded Systems Engineer in
3mdeb and open-source contributor, who has introduced us status of fwupd/LVFS
support for Qubes OS. Starting with overall informations about Firmware Update
and Linux Vendor Firmware Service, Norbert described three layers of it’s
architecture (system, session, internet) and explained how exactly works secure
web service used by hardware vendors to upload firmware archives. He also
presented fwupdmgr CLI, client tool for manual update process, describing it’s

> role between LVFS database and fwupd. Inside project status, sponsored by NLNET

Reviewer 2: @KBanecka NLnet Foundation

Author: done

foundation, we’ve heard about Qubes OS support challenges and architecture
solutions, architecture plan of the fwupd/LVFS support for Qubes OS based on
sys-usb, dom0 and UpdateVM, process of downloading and updating firmware and
what is going to be done in the next steps of development: qubes fwupdmgr script

> and .cab arhives validation. After the presentation came many questions around

Reviewer: ```suggestion
and .cab archives validation. After the presentation many questions came around
```

Author: done

> and .cab arhives validation. After the presentation came many questions around

Reviewer 3: @KBanecka maybe wrap each .cab (\`.cab\`) to get code syntax: `.cab`

Author: done

the topic. We were considering how .cab archives are signed, Qubes Dom 0 update
issues, .cab file validation tool, sevices that provide proofs of
reproducibility and many more.

> ### Different approaches of using RTMS in Qubes OS

Reviewer 2: RTMs?

Reviewer: Root of Trust for Measurement = RTM, so it should be RTMs @KBanecka

Author: done

The second day of the 'minisummit' was devoted to different approaches of using

> RTMS in Qubes OS. Piotr Król, Founder and Embedded Systems Consultant, has

Reviewer: a/a

Author: done

introduced SRTM for Qubes OS VMS, covering feasibility and security of various
S-RTM implementations for Qubes OS virtual machines. Piotr has presented S-RTM,
giving practical use cases and explained how it is created on real hardware,
presenting ways of moving it to VMs. He described Xen stub domains and explained

> how to enable TPM in QEMU S-RTM, considering options of bootfirmare,

Reviewer 2: ```suggestion
how to enable TPM in QEMU, considering options of boot firmware,
```

bootloaders, operating systems and Xen version. We were broadly familiarised
with libtpm-based TPM emulator swtpm, Xen vTPMs and finally with assumptions and
future ideas for 3mdeb. Among Q&A questions specialists were debating over an

> option for hosting QUEM emulation outside of QUEM and general idea of TPM inside

Reviewer 2: ```suggestion
option for hosting QEMU emulation outside of QEMU and a general idea of TPM inside
```

> VM.

Reviewer 2: ```suggestion
VM. 3mdeb looks for possible founding of S-RTM effort either through the foundation, commercial agreement, or community effort.
```

Michał Żygowski introduced Anti Evil Maid for modern AMD UEFI-based platform,

> where he has explained what exactly Evil Maid attacs are and how can we protect

Reviewer 2: ```suggestion
where he has explained what exactly Evil Maid attacks are and how can we protect
```

> ourselves from them with Qubes OS Ani-Evil-Maid. Michał presented current status

Reviewer: ```suggestion
ourselves from them with Qubes OS Anti-Evil-Maid. Michał presented the current status
```

Author: done

> of AEM, explained what it provides and which attacs are still not prevented.

Reviewer 2: ```suggestion
of AEM, explained what it provides and which attacks are still not prevented.
```

Author: done

Afterwards, prelegent described ways of enabling AEM in Qubes OS for AMD
platform and for TPM2 showing instalation steps, sources and repositories.
During Q&A we were considering how does it work with UEFI, what differs AMD from
Intel in AEM script and more.

### DRTM in Qubes OS – not only for DOM0 but also for VMS

> The third day was devoted to DRTM in Qubes OS – not only for DOM0 but also for

Reviewer: ```suggestion
The third day was devoted to DRTM in Qubes OS – not only for Dom0 but also for
```

Author: done

> VMS. Piotr Król has introduced DRTM for Qubes OS VMS, discussing the value and

Reviewer: ```suggestion
VMs. Piotr Król has introduced DRTM for Qubes OS VMs, discussing the value and
```

Author: done

usage models of D-RTM implementation in Qubes OS. He started presenting Root of
Trust family discussing D-RTM and how does it differ from S-RTM proceeding with
an overview of boot process for Qubes OS with Early/Late Launch scenario and

> Flicker session. Based on flicker, Piotr has proposed scenarios in which D-RTM

Reviewer 2: ```suggestion
Flicker session. Based on Flicker, Piotr has proposed scenarios in which D-RTM
```

Author: done

> may be used: Platform Relaunch, Virtual Machine Introspection technique and

Reviewer 2: ```suggestion
may be used for: Platform Relaunch, Virtual Machine Introspection technique and
```

Author: done

D-RTM, vTPM and D-RTM, Network booted vDLME and Visual Trust level indicator for

> Vms. The presentation has ended with future ideas, among others trusted system

Reviewer 2: ```suggestion
VMs. The presentation has ended with future ideas, among others trusted system
```

Author: done

backups and migration, trusted firmware update, dynamic RPC policy or secure
storage. The broad discussion covered among others the best way to re-establish

> trust in the platform.

Reviewer 2: ```suggestion
trust in the platform. D-RTM is a vast and complicated topic and requires much more education and development. Our work related to [OpenDRTM for AMD using TrenchBoot](https://nlnet.nl/project/OpenDRTM/) founded by NLnet should move the ecosystem forward, but we still looking at how to advance Intel-based solutions. There is also quite a lot of work in Xen and Linux kernel. We believe that Qubes OS minisummit 2021 should cover practical demos of TrenchBoot as reference open source D-RTM implementation.
```

Author: done

### Qubes OS testing and development – by Qubes core developers

On the last day of minisummit Qubes core developers took the floor. Frédéric
Pierret, general packaging, CentOS and Fedora templates maintainer, explained
how to build Qubes: from components to operating system overview. He desribed
what is Qubes OS composed of, in terms of developing qubes (UX, Qubes, Isolation
Provider Layer), presented the choice of Fedora as dom0 and VMs side
distributions overview. In the next step, Frederic has explained how Qubes OS
developers are introducing new features, fixes, what tools do they use and
develop, on the example of Qubes builder v.1 and v.2.

The last prelegent, Marek Marczykowski-Górecki, Project lead in QubesOS has
introduced operating system testing, when it itself uses virtual machines. Marek
briefly presented how to test Qubes OS, by describing python unit tests and
pytest frameworks, used inside Qubes OS with Travis CI. He has also mentioned
fuzzing tests, build tests, integration tests, installation tests, presented
openQA tool used for running tests in VM and described missing parts of testing
Qubes. During Q&A session we were considering how the community can help

> extending test scope and what kind of test are the most necessarry for Qubes OS

Reviewer 3: @KBanecka `necessary`

Author: done

development.

## Summary

>  Qubes OS & 3mdeb 'minisummit' has been a great event that triggered many

Reviewer: Obsolete space?

Author: deleted

important questions and allowed the community to take an active part in the
broad discussion around covered issues. We would like to thank Marek and
Frédéric for sharing your knowledge and time through all the events. Thank you
for your cooperation and willingness to ask and answer. Great appreciation goes
towards all the community that joined our meetings and many times convinced us
that events such as 'minisummit' are important and necessary. We have exchanged
loads of knowledge and experience and we would like to meet again, on Qubes OS
and 3mdeb 'minisummit' 2021. Thank you all once again.

>  If you think we can help in improving Qubes OS support for your hardware, help

Reviewer: a/a

Author: I would rather leave this space here

Reviewer: Why? I don't recall any blog post having the space here
